src/main/java/io/dapr/components/cli/PluggableComponentServer.java

```diff
@@ -0,0 +1,168 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components.cli;
+
+import com.beust.jcommander.JCommander;
+import io.grpc.BindableService;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.netty.NettyServerBuilder;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerDomainSocketChannel;
+import io.netty.channel.unix.DomainSocketAddress;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.java.Log;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wraps a GRPC service exposing a Dapr pluggable component into a fully functional executable.
+ *
+ * <p>Takes care of:
+ * 1. CLI argument parsing,
+ * 2. setting up Unix Domain Socket files,
+ * 3. environment variable handling,
+ * 4. setting up the required server machinery to expose the service and handle requests for it.
+ *
+ * <p>If you end up needing to extend functionality of this class it might be simpler re-implementing it.
+ */
+@Log
+@RequiredArgsConstructor
+public class PluggableComponentServer {
+  /**
+   * Nane of the current CLI program. This will be used for constructing our --help messages.
+   */
+  @NonNull
+  private String programName;
+
+  /**
+   * GRPC service for the Dapr component you want to have exposed.
+   */
+  @NonNull final BindableService exposedService;
+
+  // Initialized by run, which is the only public method in this class.
+  private Server server;
+
+  /** Setups the machinery and wrappers to expose your service as a server.
+   *
+   * <p>Takes care of setting up a server and parsing command line to serve your
+   * service.</p>
+   *
+   * <p>Ideally a server can only expose/serve a single service. If you don't
+   * know what you are doing, do no try to expose more than once service / component
+   * with the same server. While possible, this is not the model that Dapr itself
+   * expects from components and it won't know how to handle multiple components
+   * being served from the same UNIX socket.</p>
+   *
+   * @param args Command line arguments that needs to be parsed.
+   *
+   * @throws IOException Thrown if an error happened while running the service.
+   * @throws InterruptedException Thrown if an error happened while running the service.
+   */
+  public void main(@NonNull final String[] args) throws IOException, InterruptedException {
+    // Command line parsing
+    final CommandLineOptions options = new CommandLineOptions();
+    final JCommander jCommander = JCommander.newBuilder()
+        .programName(programName)
+        .addObject(options)
+        .build();
+    jCommander.parse(args);
+    if (options.isHelp()) {
+      jCommander.usage();
+      System.exit(0);
+    }
+    // Either Unix Socket Domains or TCP, but not both
```

Comment:
I agree with you that's good for testing purposes but if we're going to provide a way to use it without a unix socket things should change on the dapr runtime as well.

I would remove this code just to make sure we don't create wrong expectations from the user

Reply:
Fixed as of 68ba01b44035a42884454d93df2e179f598a39c5